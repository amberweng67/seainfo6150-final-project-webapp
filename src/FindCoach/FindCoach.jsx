import React from "react";
import styles from './FindCoach.module.css';
import CoachCard from '../CoachCard/CoachCard';
import useCoaches from '../Hooks/useCoaches';

const FindCoach = () => {
  const [coaches] = useCoaches();
  if (!coaches) {
    return null;
  }
  if (coaches.length === 0) {
    return 'There is no coach data right now';
  }

  return (
    <div className={styles.container}>
      <h2>Find a coach</h2>
      <div className={styles.listContainer}>
        {coaches.map(card => <CoachCard key={card.id} card={card} />)}
      </div>
    </div>  
  );
};


export default FindCoach;
