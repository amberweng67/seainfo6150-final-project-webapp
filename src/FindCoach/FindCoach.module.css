.listContainer {
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  gap: 10px;
}

@media screen and (max-width: 600px) {
  .listContainer {
    grid-template-columns: repeat(1, 1fr);
  }
}

@media screen and (max-width: 960px) and (min-width:601px) {
  .listContainer {
    grid-template-columns: repeat(3, 1fr);
  }
}

/* .listContainer > *:not(:last-child) {
  margin-right: 16px;
  margin-bottom: 16px;
} */

.container {
  padding: 20px;
}