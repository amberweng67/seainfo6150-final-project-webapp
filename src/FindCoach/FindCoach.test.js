import React from "react";
import FindCoach from "./FindCoach.jsx";

it("renders the find coach component correctly", () => {
  const { container } = render(<FindCoach />);
  expect(container).toMatchSnapshot();
});
