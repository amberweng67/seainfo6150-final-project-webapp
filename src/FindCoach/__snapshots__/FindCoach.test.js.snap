// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the find coach component correctly 1`] = `<div />`;
